iroh/tests/gc.rs

```diff
@@ -205,74 +217,84 @@ mod flat {
         // this assumes knowledge of the internal directory structure of the flat store
-        path(root.join("complete"), "data")
+        path(root.join("data"), "data")
     }
 
     fn outboard_path(root: PathBuf) -> impl Fn(&iroh_bytes::Hash) -> PathBuf {
         // this assumes knowledge of the internal directory structure of the flat store
-        path(root.join("complete"), "obao4")
+        path(root.join("data"), "obao4")
     }
 
-    async fn sync_directory(dir: impl AsRef<Path>) -> io::Result<()> {
-        // sync the directory to make sure the metadata is written
-        // does not work on windows
-        if let Ok(dir) = std::fs::File::open(dir) {
-            dir.sync_all().ok();
-        }
-        tokio::time::sleep(Duration::from_millis(50)).await;
-        Ok(())
+    async fn check_consistency(store: &impl Store) -> anyhow::Result<ValidateLevel> {
+        let mut max_level = ValidateLevel::Trace;
+        let (tx, mut rx) = tokio::sync::mpsc::channel(1);
+        let task = tokio::task::spawn(async move {
+            while let Some(ev) = rx.recv().await {
+                if let ValidateProgress::ConsistencyCheckUpdate { level, .. } = &ev {
+                    max_level = max_level.max(*level);
+                }
+            }
+        });
+        store.validate(tx).await?;
+        task.await?;
+        Ok(max_level)
     }
 
-    /// count the number of partial files for a hash. partial files are <hash>-<uuid>.<suffix>
-    fn count_partial(
-        root: PathBuf,
-        suffix: &'static str,
-    ) -> impl Fn(&iroh_bytes::Hash) -> std::io::Result<usize> {
-        move |hash| {
-            let valid_names = std::fs::read_dir(&root)?
-                .filter_map(|e| e.ok())
-                .filter_map(|e| {
-                    if e.metadata().ok()?.is_file() {
-                        e.file_name().into_string().ok()
-                    } else {
-                        None
-                    }
-                });
-            let prefix = format!("{}-", hash.to_hex());
-            Ok(valid_names
-                .filter(|x| x.starts_with(&prefix) && x.ends_with(suffix))
-                .count())
+    #[tokio::test]
+    async fn redb_doc_import_stress() -> Result<()> {
+        let _ = tracing_subscriber::fmt::try_init();
+        let dir = testdir!();
+        let bao_store = iroh_bytes::store::redb::Store::load(dir.join("store")).await?;
+        let node = wrap_in_node(bao_store.clone(), Duration::from_secs(10)).await;
+        let client = node.client();
+        let doc = client.docs.create().await?;
+        let author = client.authors.create().await?;
+        let temp_path = dir.join("temp");
+        std::fs::create_dir_all(&temp_path)?;
+        let mut to_import = Vec::new();
+        for i in 0..10000 {
+            let data = create_test_data(16 * 1024 * 3 + 1);
+            let path = temp_path.join(format!("file{}", i));
+            std::fs::write(&path, &data)?;
```

Comment:
please use `async` methods if you are inside an async function :)

Reply:
I hate them but OK.